File: app/src/main/java/com/project/scratchstudio/kith_andoid/UI/BoardList/list/BoardHolder.java
package com.project.scratchstudio.kith_andoid.UI.BoardList.list;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageView;

import com.project.scratchstudio.kith_andoid.CustomViews.CustomFontTextView;
import com.project.scratchstudio.kith_andoid.R;
import com.project.scratchstudio.kith_andoid.network.model.board.Board;


public class BoardHolder extends RecyclerView.ViewHolder  {

    public ImageView image;
    public View bluring;
    public CustomFontTextView title;
    public CustomFontTextView date;
    public CheckBox favorite;

    public BoardHolder(@NonNull View itemView){
        super(itemView);
        image = itemView.findViewById(R.id.photo);
        bluring = itemView.findViewById(R.id.view);
        title = itemView.findViewById(R.id.title);
        date = itemView.findViewById(R.id.date);
        favorite = itemView.findViewById(R.id.favorite);
    }

    public void bind(final Board item, final BoardAdapter.OnItemClickListener listener, int id) {
        itemView.setOnClickListener(v -> listener.onItemClick(item, id, this));
    }
}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/ui/BoardList/list/BoardHolder.kt
package com.project.scratchstudio.kith_andoid.ui.BoardList.list

import android.view.View
import android.widget.CheckBox
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.project.scratchstudio.kith_andoid.CustomViews.CustomFontTextView
import com.project.scratchstudio.kith_andoid.R
import com.project.scratchstudio.kith_andoid.network.model.board.Board

class BoardHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    var image: ImageView = itemView.findViewById(R.id.photo)
    var title: CustomFontTextView = itemView.findViewById(R.id.title)
    var date: CustomFontTextView = itemView.findViewById(R.id.date)
    var favorite: CheckBox = itemView.findViewById(R.id.favorite)

    fun bind(item: Board, listener: BoardAdapter.OnItemClickListener, id: Int) {
        itemView.setOnClickListener { v -> listener.onItemClick(item, id, this) }
    }
}
